ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/RemoteConnectorExecutor.java

```diff
@@ -188,16 +200,78 @@ && getUserRateLimiterMap().get(user.getName()) != null
             if (getMlGuard() != null && !getMlGuard().validate(payload, MLGuard.Type.INPUT)) {
                 throw new IllegalArgumentException("guardrails triggered for user input");
             }
-            invokeRemoteModel(mlInput, parameters, payload, tensorOutputs, countDownLatch, actionListener);
+            if (getConnectorClientConfig().getMaxRetryTimes() != 0) {
+                invokeRemoteModelWithRetry(mlInput, parameters, payload, executionContext, actionListener);
+            } else {
+                invokeRemoteModel(mlInput, parameters, payload, executionContext, actionListener);
+            }
+        }
+    }
+
+    default BackoffPolicy getRetryBackoffPolicy(ConnectorClientConfig connectorClientConfig) {
+        // we've validated the value of retryBackOffPolicy at ConnectorClientConfig.parse
+        switch (connectorClientConfig.getRetryBackoffPolicy()) {
+            case "exponential_equal_jitter":
+                return BackoffPolicy
+                    .exponentialEqualJitterBackoff(
+                        connectorClientConfig.getRetryBackoffMillis(),
+                        connectorClientConfig.getRetryTimeoutSeconds()
+                    );
+            case "exponential_full_jitter":
+                return BackoffPolicy.exponentialFullJitterBackoff(connectorClientConfig.getRetryBackoffMillis());
+            default:
+                return BackoffPolicy
+                    .constantBackoff(TimeValue.timeValueMillis(connectorClientConfig.getRetryBackoffMillis()), Integer.MAX_VALUE);
```

Comment:
What's the second parameter and why using Integer.MAX_VALUE, please add some comment.

Reply:
The second parameter is the maxNumberOfRetries for ConstantBackoff. However, we can't reuse it, because the ConstantBackoffIterator.next() throws exception when it reaches the limit, but the RetryableAction doesn't handle the exception throwed from iterator, and will make the request hanging. 

Setting it to  Integer.MAX_VALUE to avoid throwing this exception. Instead we handle the max retry numbers at `shouldRetry`.

Reply:
Comments added